entity-framework/core/modeling/value-conversions.md

```diff
@@ -2,145 +2,746 @@
+
+Value converters can be used in two ways to deal with this. First, EF Core has a value converter that creates an 8-byte opaque value which preserves the `Kind` flag. For example:
+
+<!--
+                modelBuilder.Entity<Post>()
+                    .Property(e => e.PostedOn)
+                    .HasConversion<long>();
+-->
+[!code-csharp[ConfigurePreserveDateTimeKind1](../../../samples/core/Modeling/ValueConversions/PreserveDateTimeKind.cs?name=ConfigurePreserveDateTimeKind1)]
+
+This allows DateTime values with different `Kind` flags to be mixed in the database.
+
+The problem with this approach is that the database no longer has recognizable `datetime` or `datetime2` columns. So instead it is common to always store UTC time (or, less commonly, always local time) and then either ignore the `Kind` flag or set it to the appropriate value using a value converter. For example, the converter below ensures that the `DateTime` value read from the database will have the <xref:System.DateTimeKind> `UTC` kind:
+
+<!--
+                modelBuilder.Entity<Post>()
+                    .Property(e => e.LastUpdated)
+                    .HasConversion(
+                        v => v,
+                        v => new DateTime(v.Ticks, DateTimeKind.Utc));
+-->
+[!code-csharp[ConfigurePreserveDateTimeKind2](../../../samples/core/Modeling/ValueConversions/PreserveDateTimeKind.cs?name=ConfigurePreserveDateTimeKind2)]
+
+### Use case-insensitive string keys
+
+Some databases, including SQL Server, perform case-insensitive string comparisons by default. .NET on the other hand performs case-sensitive string comparisons by default. This means that a foreign key value like "DotNet" will match the primary key value "dotnet" on SQL Server, but will not match it in EF Core. A value comparer for keys can be used for force EF Core into case-insensitive string comparisons like in the database. For example, consider a blog/posts model with string keys:
+
+<!--
+        public class Blog
+        {
+            public string Id { get; set; }
+            public string Name { get; set; }
+
+            public ICollection<Post> Posts { get; set; }
+        }
+
+        public class Post
+        {
+            public string Id { get; set; }
+            public string Title { get; set; }
+            public string Content { get; set; }
+
+            public string BlogId { get; set; }
+            public Blog Blog { get; set; }
+        }
+-->
+[!code-csharp[CaseInsensitiveStringsModel](../../../samples/core/Modeling/ValueConversions/CaseInsensitiveStrings.cs?name=CaseInsensitiveStringsModel)]
+
+This will not work as expected if some of the `Post.BlogId` values have different casing. A value comparer can be used to correct this:
```

Comment:
It may be good to add a sentence detailing exactly how this will fail if the comparer isn't defined, to make users better understand why this is important.